core/src/main/java/org/infinispan/util/concurrent/locks/DeadlockDetectingLockManager.java

```diff
@@ -46,77 +50,62 @@
-               if (trace) log.tracef("Could not acquire lock as %s is locked by %s (%s)", key, owner, System.identityHashCode(owner));
-               if (isDeadlockAndIAmLoosing(lockOwnerTx, thisTx, key)) {
-                  updateStats(thisTx);
-                  String message = String.format("Deadlock found and we (%s) shall not continue. Other tx is %s",
-                                                 thisTx, lockOwnerTx);
-                  log.trace(message);
-                  throw new DeadlockDetectedException(message);
-               }
-            }
-         }
-      } else {
-         return super.lockAndRecord(key, ctx, lockTimeout);
+   public KeyAwareLockPromise lock(Object key, Object lockOwner, long time, TimeUnit unit) {
+      if (lockOwner instanceof DldGlobalTransaction) {
+         ((DldGlobalTransaction) lockOwner).setLockIntention(Collections.singleton(key));
+         KeyAwareLockPromise promise = super.lock(key, lockOwner, time, unit);
+         promise.setDeadlockChecker(this);
+         return promise;
       }
-      // couldn't acquire lock!
-      return false;
+      return super.lock(key, lockOwner, time, unit);
+   }
+
+   @Override
+   public KeyAwareLockPromise lockAll(Collection<?> keys, Object lockOwner, long time, TimeUnit unit) {
+      if (lockOwner instanceof DldGlobalTransaction) {
+         ((DldGlobalTransaction) lockOwner).setLockIntention(new HashSet<>(keys));
+         KeyAwareLockPromise promise = super.lockAll(keys, lockOwner, time, unit);
+         promise.setDeadlockChecker(this);
+         return promise;
+      }
+      return super.lockAll(keys, lockOwner, time, unit);
    }
 
-   private boolean isDeadlockAndIAmLoosing(DldGlobalTransaction lockOwnerTx, DldGlobalTransaction thisTx, Object key) {
+   private boolean isDeadlockAndIAmLoosing(DldGlobalTransaction lockOwnerTx, DldGlobalTransaction thisTx) {
       //run the lose check first as it is cheaper
       boolean wouldWeLoose = thisTx.wouldLose(lockOwnerTx);
       if (!wouldWeLoose) {
-         if (trace) log.tracef("We (%s) win against the other (%s) transaction, so no point running rest of DLD", thisTx, lockOwnerTx);
+         if (trace)
+            log.tracef("We (%s) win against the other (%s) transaction, so no point running rest of DLD", thisTx, lockOwnerTx);
          return false;
       }
       //do we have lock on what other tx intends to acquire?
-      return ownsLocalIntention(thisTx, lockOwnerTx.getLockIntention()) || ownsRemoteIntention(lockOwnerTx, thisTx, key) || isSameKeyDeadlock(key, thisTx, lockOwnerTx);
+      return ownsAnyLocalIntention(thisTx, lockOwnerTx) || ownsRemoteIntention(lockOwnerTx, thisTx) || isSameKeyDeadlock(thisTx, lockOwnerTx);
    }
 
-   private boolean isSameKeyDeadlock(Object key, DldGlobalTransaction thisTx, DldGlobalTransaction lockOwnerTx) {
+   private boolean isSameKeyDeadlock(DldGlobalTransaction thisTx, DldGlobalTransaction lockOwnerTx) {
```

Comment:
I think this method can be removed, it was only necessary when we were acquiring the lock on the originator/backup owners.